### PR TITLE
New Protocol - HT6P20B

### PR DESCRIPTION
**Pull request by ricardojlrufino**
New method added
mySwitch.enableReceiveProtocol(RCSWITCH_HT6P20B);

Small refactoring in struct Protocol

### CODE REVIEW

**Comment by ricardojlrufino** [comment]
Only RECEIVE TESTED

**Comment by fingolfin** [comment]
Please explain the logic behind the new `Protocol::startData` -- what is it good for, why is it necessary?

File: RCSwitch.cpp [hunk 1]
```diff
@@ -45,6 +46,7 @@ unsigned long RCSwitch::nReceivedValue = 0;
 unsigned int RCSwitch::nReceivedBitlength = 0;
 unsigned int RCSwitch::nReceivedDelay = 0;
 unsigned int RCSwitch::nReceivedProtocol = 0;
+unsigned int RCSwitch::enabledProtocols = B00000111;
```
Comment (fingolfin) [inline RCSwitch.cpp]: The name of this member variable is misleading -- a better one would be `enabledReceiveProtocols` or `enabledReceptionProtocols` or so

File: RCSwitch.cpp [hunk 2]
```diff
@@ -583,6 +596,9 @@ void RCSwitch::enableReceive() {
   if (this->nReceiverInterrupt != -1) {
     RCSwitch::nReceivedValue = 0;
     RCSwitch::nReceivedBitlength = 0;
+    #if defined(CORE_TEENSY)
+      pinMode(this->nReceiverInterrupt, INPUT);
+    #endif
     attachInterrupt(this->nReceiverInterrupt, handleInterrupt, CHANGE);
```
Comment (fingolfin) [inline RCSwitch.cpp]: This change is unrelated to this pull request -- please either remove it here and put it into a separate pull request, or exaplin why it must be mixed with this pull request.

File: RCSwitch.h [hunk 3]
```diff
@@ -48,6 +48,13 @@
 // We can handle up to (unsigned long) => 32 bit * 2 H/L changes per bit + 2 for sync
 #define RCSWITCH_MAX_CHANGES 67
 
+#define RCSWITCH_PROTOCOL_1 1
+#define RCSWITCH_PROTOCOL_2 2
+#define RCSWITCH_PROTOCOL_3 3
+#define RCSWITCH_PROTOCOL_4 4
+#define RCSWITCH_PROTOCOL_5 5
+#define RCSWITCH_HT6P20B 6
+
```
Comment (fingolfin) [inline RCSwitch.h]: These new #defines are also not mentioned in the description of the pull request.

I guess it makes sense to have symbolic names for the predefines protocols, and encourage people to use them. But then this should be done by defininig an enum, and `setProtocol` could be changed to take such an enum value as argument.

File: RCSwitch.h [hunk 4]
```diff
@@ -97,14 +104,16 @@ class RCSwitch {
 
     struct Protocol {
         int pulseLength;
+        int startData;
         HighLow syncFactor;
         HighLow zero;
         HighLow one;
+        bool enabled;
```
Comment (fingolfin) [inline RCSwitch.h]: Why is this `enabled` member there? It seems to be unused -- perhaps a left-over from an earlier revision of the PR?

**Comment by fingolfin** [comment]
This pull request is currently not mergable, and the submitter, @ricardojlrufino , did not react to my questions, so I am closing this for now. Feel free to open a new pull request if you are still interested, or reopen this one.
